woocommerce/woocommerce-android
# Issue/2844 order detail refactor order status card

This PR takes care of the first step of the Order Detail refactoring #2844 . 

####  Changes
- All the order detail changes are moved into the `orders/details` package. 
- Adds new `OrderDetailViewModel`, `OrderDetailRepository`, `OrderDetailFragment` and `OrderDetailModule`.
- Adds new package `views` to hold all the custom views used by the Order detail screen. 
- Adds the Order Detail Status Info card (`OrderDetailOrderStatusView`) to the Order detail screen.

#### Notes
- I have split this massive change into multiple small PRs in order to make reviews easier. 
- This PR is to be merged into a feature branch.
- This PR just adds the Order detail status card to the Order detail screen. The remaining cards will be added in subsequent PRs along with unit tests and click actions for all the buttons in the Order detail.

#### Testing
- Click on an order from the `Orders` tab.
- Notice the toolbar title displays the Order number correctly.
- Notice the Order number, date, name and status is displayed correctly. 

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/93462740-68b37c00-f904-11ea-8fb2-41448d060765.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/93462797-7f59d300-f904-11ea-9028-c275235ab098.png" width="300"/>

Update release notes:
